tests/Tests/500__jupyterhub/custom-image.robot

```diff
@@ -0,0 +1,42 @@
+*** Settings ***
+Documentation    Testing the backend for custom images (Adding ImageStream to redhat-ods-applications)
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/GPU.resource
+Library          JupyterLibrary
+Library          OpenShiftCLI
+Suite Teardown   Custom Image Teardown
+
+
+*** Variables ***
+${YAML} =         configs/resources/custom_image.yaml
+${IMG_NAME} =     custom-test-image
+
+
+*** Test Cases ***
+Verify Custom Image Can Be Added
+    [Documentation]    Applies the YAML and Gets the ImageStream
+    ...                Then loads the spawner and tries using the custom img
+    [Tags]    Tier2
+    ...       ODS-1208
+    ${apply_result} =    Run Keyword And Return Status    Run    oc apply -f ${YAML}
+    Should Be Equal    "${apply_result}"    "True"
+    # OpenShiftCLI.Apply    kind=ImageStream    src="configs/resources/custom_image.yaml"
+    # ...    namespace=redhat-ods-applications
+    ${get_metadata} =    OpenShiftCLI.Get    kind=ImageStream    field_selector=metadata.name==${IMG_NAME}
+    ...    namespace=redhat-ods-applications
+    &{data} =    Set Variable    ${get_metadata}[0]
+    Should Be Equal    ${data.metadata.name}    ${IMG_NAME}
```

Comment:
I'd write a new keyword `Get Image Stream Metadata` which takes as args the field you want (name), the image stream, namespace and returns the metadata value. 

Or as alternative, a new keyword to wrap lines 28-30-32

Reply:
The problem with parametrizing `the field you want (name)` is that traversing the output is not straightforward and I usually need to check manually how to get to a specific field. I can hardcode the name filter, but it'd be hard to do it for any possible field.

Reply:
I believe @pablofelix has implemented a filter functionality in the latest version of openshiftcli which is able to traverse the output and give you any field you want, but we still haven't updated ods-ci to use. Perhaps it is time to move to the new version? Then this problem would be solved.

Will wrap this in a keyword for now

Reply:
Done in latest commit